src/XMakeBuildEngine/BackEnd/Components/Logging/EventSourceSink.cs

```diff
@@ -308,6 +354,9 @@ private void RaiseMessageEvent(object sender, BuildMessageEventArgs buildEvent)
         /// <exception cref="Exception">ExceptionHandling.IsCriticalException exceptions will not be wrapped</exception>
         private void RaiseErrorEvent(object sender, BuildErrorEventArgs buildEvent)
         {
+            // Keep track of build submissions that have logged errors.  If there is no build context, add a submission ID of "-1"
+            BuildSubmissionIdsThatHaveLoggedErrors.Add(buildEvent != null && buildEvent.BuildEventContext != null ? buildEvent.BuildEventContext.SubmissionId : -1);
```

Comment:
Great suggestion, I looked at it and ReSharper didn't suggest this so I just moved on.  I will update.